zebra-network/src/peer_set/candidate_set/tests/prop.rs

```diff
@@ -63,33 +112,60 @@ proptest! {
             // Check rate limiting for initial peers
             check_candidates_rate_limiting(&mut candidate_set, initial_candidates).await;
             // Sleep more than the rate limiting delay
-            sleep(Duration::from_millis(400)).await;
+            sleep(MAX_TEST_CANDIDATES * MIN_PEER_CONNECTION_INTERVAL).await;
             // Check that the next peers are still respecting the rate limiting, without causing a
             // burst of reconnections
             check_candidates_rate_limiting(&mut candidate_set, extra_candidates).await;
         };
 
-        assert!(runtime.block_on(timeout(Duration::from_secs(10), checks)).is_ok());
+        // Allow enough time for the maximum number of candidates,
+        // plus some extra time for test machines with high CPU load.
+        // (The max delay asserts usually fail before hitting this timeout.)
+        let max_rate_limit_sleep = 3 * MAX_TEST_CANDIDATES * MIN_PEER_CONNECTION_INTERVAL;
+        let max_extra_delay = (2 * MAX_TEST_CANDIDATES + 1) * MAX_SLEEP_EXTRA_DELAY;
+        assert!(runtime.block_on(timeout(max_rate_limit_sleep + max_extra_delay, checks)).is_ok());
```

Comment:
Increase the test timeout to match any increases in the candidates or times.
(Currently it's 10.2 seconds rather than 10 seconds!)